fltk: Fl_JPEG_Image.cxx bug in fltk-1.3.x for jpeg-9

Link: http://www.fltk.org/str.php?L2920
Version: 1.3-current

There has been a change in jpeg-9:

Fl_JPEG_Image.cxx: In constructor 'Fl_JPEG_Image::Fl_JPEG_Image(const
char*)':
Fl_JPEG_Image.cxx:158:29: error: invalid conversion from 'int' to
'boolean' [-fpermissive]
In file included from Fl_JPEG_Image.cxx:50:0:
/usr/include/jpeglib.h:1032:13: error:   initializing argument 2 of 'int
jpeg_read_header(j_decompress_ptr, boolean)' [-fpermissive]


http://permalink.gmane.org/gmane.comp.lib.fltk.bugs/10893

Upstream-Status: Backport

partical back port as this version is only affected by on change.

Signed-off-by: Armin Kuster <akuster808@gmail.com>

Index: fltk-1.1.10/src/Fl_JPEG_Image.cxx
===================================================================
--- fltk-1.1.10.orig/src/Fl_JPEG_Image.cxx
+++ fltk-1.1.10/src/Fl_JPEG_Image.cxx
@@ -153,7 +153,7 @@ Fl_JPEG_Image::Fl_JPEG_Image(const char
 
   jpeg_create_decompress(&dinfo);
   jpeg_stdio_src(&dinfo, fp);
-  jpeg_read_header(&dinfo, 1);
+  jpeg_read_header(&dinfo, TRUE);
 
   dinfo.quantize_colors      = (boolean)FALSE;
   dinfo.out_color_space      = JCS_RGB;
